app/models/spree/flow_io_product_decorator.rb

```diff
@@ -40,22 +40,34 @@ def price_range(product_zone)
             price = v.price_in(currency)
             next if price.nil? || price.amount.nil?
 
-            min = price if min.nil? || min.amount > price.amount
-            max = price if max.nil? || max.amount < price.amount
+            min = calculate_new_min(price, min)
+            max = calculate_new_max(price, max)
```

Comment:
Maybe something like this could be useful?
```suggestion
            min = [price, min].compact.min
            max = [price, max].compact.max
```

Reply:
Sure! Changed!